ruma-events/src/room/relationships.rs

```diff
@@ -0,0 +1,161 @@
+//! Types for event relationships.
+//!
+//! Events in Matrix can relate to one another in a couple of ways, this module
+//! adds types to parse the relationship of an event if any exists.
+//!
+//! MSC for all the relates_to types except replies:
+//!     https://github.com/matrix-org/matrix-doc/pull/2674
+
+use ruma_identifiers::EventId;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+
+/// Enum modeling the different ways relationships can be expressed in a
+/// `m.relates_to` field of an event.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum RelatesTo {
+    /// A relation which contains subtypes indicating the type of the
+    /// relationship with the `rel_type` field.
+    Relation(Relation),
+    /// An `m.in_reply_to` relationship indicating that the event is a reply to
+    /// another event.
+    Reply(Reply),
```

Comment:
No need for `Reply` to be a separate type I think.

```suggestion
    Reply {
	    /// Information about another message being replied to.
	    #[serde(rename = "m.in_reply_to")]
	    pub in_reply_to: InReplyTo,
	}
```

Reply:
We could even consider changing `in_reply_to: InReplyTo` to `in_reply_to: EventId`, removing one level of nesting present in the JSON. Of course that's a bit risky because the spec could introduce new fields both next to or inside `in_reply_to`.